### PR TITLE
Add FFI generator for Windows

### PR DESCRIPTION
**Pull request by rfl890**
Uses [`RtlGenRandom`](https://learn.microsoft.com/en-us/windows/win32/api/ntsecapi/nf-ntsecapi-rtlgenrandom) API call to securely generate random data in a vanilla LuaJIT environment.

### CODE REVIEW

File: src/uuid/rng/init.lua [hunk 1]
```diff
@@ -78,7 +78,42 @@ function rng.luasystem()
...
+      ok, ffi = pcall(require, "cffi")
+  end
+  if not ok then
+    return nil, "ffi not available"
+  end
 
+  local ffi_defs = [[
+    unsigned char SystemFunction036(void *RandomBuffer, unsigned long RandomBufferLength);
+  ]]
+  ffi.cdef(ffi_defs)
```
Comment (Tieske) [inline src/uuid/rng/init.lua:101]: this will probably fail if you call it again

**Review by Tieske — DISMISSED** [review]
Also in the code for generating a seed, this one should be tried

**Review by Tieske — CHANGES_REQUESTED** [review]
Also in the code for generating a seed, this one should be tried

File: src/uuid/rng/init.lua (hunk 1 above)
Comment (rfl890) [reply src/uuid/rng/init.lua:101]: This testing code shows it simply appends to the definitions:
```lua
do
    local ffi = require("ffi")
    ffi.cdef[[
        void rand();
    ]]
end
do
    local ffi = require("ffi")
    ffi.cdef[[
        void memcpy();
    ]]
end
local ffi = require("ffi")
print(ffi.C.rand)
print(ffi.C.memcpy)
```

Comment (rfl890) [reply src/uuid/rng/init.lua:101]: although it's not great to keep re-defining the same symbol over and over again, maybe some caching could be done.

File: src/uuid/rng/init.lua [hunk 2]
```diff
@@ -78,7 +78,42 @@ function rng.luasystem()
...
+-- a good source of randomness on all modern versions of Windows.
+-- Requires LuaJIT or a LuaJIT-compatible FFI library.
+-- @treturn function A function that returns `n` random bytes, signature: `byte_string, err = func(n)`
+-- @usage
+-- local uuid = require "uuid"
+-- uuid.set_rng(uuid.rng.win_ffi())
+function rng.win_ffi()
+  local ok, ffi = pcall(require, "ffi")
+  if not ok then 
+      ok, ffi = pcall(require, "cffi")
```
Comment (Tieske) [inline src/uuid/rng/init.lua:92]: which ffi module is this, can we add it to the docs?

Comment (rfl890) [reply src/uuid/rng/init.lua:92]: "ffi" is luajit's default ffi module, "cffi" is [cffi-lua](https://github.com/q66/cffi-lua), a LuaJIT-compatible ffi library.

File: src/uuid/rng/init.lua [hunk 3]
```diff
@@ -78,7 +78,42 @@ function rng.luasystem()
...
+----------------------------------------------------------------------------
+-- Returns an rng that implements Windows' RtlGenRandom function which is
+-- a good source of randomness on all modern versions of Windows.
+-- Requires LuaJIT or a LuaJIT-compatible FFI library.
+-- @treturn function A function that returns `n` random bytes, signature: `byte_string, err = func(n)`
+-- @usage
+-- local uuid = require "uuid"
+-- uuid.set_rng(uuid.rng.win_ffi())
+function rng.win_ffi()
+  local ok, ffi = pcall(require, "ffi")
```
Comment (Tieske) [inline src/uuid/rng/init.lua:90]: I think the first check should be if we're running Windows. Not loading any ffi defs if we're not going to use them anyway

File: src/uuid/rng/init.lua (hunk 2 above)
Comment (Tieske) [reply src/uuid/rng/init.lua:92]: yes, a link would be nice, the LuaJIT one is well known, but the others less so

File: src/uuid/rng/init.lua (hunk 1 above)
Comment (Tieske) [reply src/uuid/rng/init.lua:101]: yes caching such that we define it only once would be great

**Review by Tieske — CHANGES_REQUESTED** [review]

**Review by Tieske — APPROVED** [review]

**Pull request merged**